src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((acc, item) => {
+    const keyValue = item[key] as unknown as string;
```

Comment:
In this example it  works but i think T should extend somethig that has only string and numbers as values so it is impossible to use groupByKey with wrong objects T.